src/frontend/src/optimizer/mod.rs

```diff
@@ -459,6 +452,13 @@ impl PlanRoot {
             ApplyOrder::TopDown,
         );
 
+        plan = self.optimize_by_rules(
+            plan,
+            "Agg on Index".to_string(),
+            vec![TopNOnIndexRule::create()],
```

Comment:
Should we rewrite the rule for `BatchXxx` plans, or simply moving here is enough? cc @chenzl25

Reply:
Here is enough.